packages/context/src/inject.ts

```diff
@@ -165,6 +202,22 @@ function resolveAsSetter(ctx: Context, injection: Injection) {
   };
 }
 
+function resolveAsOption(ctx: Context, injection: Injection) {
+  if (injection.binding) {
```

Comment:
```ts
if (!injection.binding) return undefined
```

This makes the code easier to read and keeps our indentation levels low.

Is it actually a valid situation that `injection` does not have any `binding` specified? I think it's not, in which case we should throw an error instead of silently returning `undefined`.